client/src/store.js

```diff
@@ -20,10 +20,31 @@ export default new Vuex.Store({
   },
   actions: {
     async fetchPipelines({ commit, state }) {
-      if (state.pipelines.length === 0) {
+      if (Object.keys(state.pipelines).length === 0) {
         const { data } = await getPipelineList();
-        commit('setPipelines', data);
-        return data;
+        // Sort list of pipelines in each category by name
+        Object.keys(data).forEach((key) => {
+          const category = data[key];
+          if (category.pipes.length > 0) {
+            category.pipes.sort((a, b) => {
+              if (a.name && b.name) {
```

Comment:
Oh, last thought: this line seems overly defensive to me.  If a server returned a pipe item without the `name` property, that would be malformed data.  It's better to error on access of an expected property than have something weird happen.

For example: later, we change `name` to `pipename`.  We remember to change the template in `RunPipelineMenu.vue` but forget to change it here, so they show up but arent sorted.  What happened?!

See also: https://blog.vcarl.com/overly-defensive-programming/

Reply:
This is a good point and even related to my `category.pipes.length >0` which was checking to make sure that pipes was an array before running a sort on it.